src/auth/ejabberd_auth.erl

```diff
@@ -328,6 +333,21 @@ does_stored_user_exist(HostType, #jid{luser = LUser, lserver = LServer}) ->
 does_stored_user_exist(_HostType, error) ->
     false.
 
+-spec does_user_exist(Status :: mongoose_hooks:simple_acc(),
+                      HostType :: mongooseim:host_type(),
+                      Jid :: jid:jid()) -> does_user_exist().
+does_user_exist(#{result := true} = Status, _, _) ->
+    Status;
+does_user_exist(#{type := stored} = Status, HostType, Jid) ->
```

Comment:
it seems that `type` should be a hook argument, not an acc field. we are not supposed to change it.

Reply:
It's not changing it, it's matching on it only.

Reply:
so that's an argument, not a part of the Acc.